controlplane/kubeadm/internal/controllers/controller_test.go

```diff
@@ -1227,88 +1257,471 @@ kubernetesVersion: metav1.16.1`,
+	g.Expect(updatedInfraMachine.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:spec"}))
+
+	// The KubeadmConfig should have ownership of "labels" and "annotations" transferred to
+	// "capi-kubeadmcontrolplane" manager.
+	updatedKubeadmConfig := existingKubeadmConfig.DeepCopy()
+	g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(updatedKubeadmConfig), updatedKubeadmConfig))
+
+	// Verify ManagedFields
+	g.Expect(updatedKubeadmConfig.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(kcpManagerName, metav1.ManagedFieldsOperationApply, contract.Path{"f:metadata", "f:labels"}))
+	g.Expect(updatedKubeadmConfig.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(kcpManagerName, metav1.ManagedFieldsOperationApply, contract.Path{"f:metadata", "f:annotations"}))
+	g.Expect(updatedKubeadmConfig.GetManagedFields()).ShouldNot(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:metadata", "f:labels"}))
+	g.Expect(updatedKubeadmConfig.GetManagedFields()).ShouldNot(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:metadata", "f:annotations"}))
+	g.Expect(updatedKubeadmConfig.GetManagedFields()).Should(
+		ssa.MatchFieldOwnership(classicManager, metav1.ManagedFieldsOperationUpdate, contract.Path{"f:spec"}))
+
+	//
+	// Verify In-place mutating fields
+	//
+
+	// Update KCP and verify the in-mutating fields are propagated.
+	kcp.Spec.MachineTemplate.ObjectMeta.Labels = map[string]string{
+		"preserved-label": "preserved-value",  // Keep the label and value as is
+		"modified-label":  "modified-value-2", // Modify the value of the label
+		// Drop "dropped-label"
+	}
+	expectedLabels := map[string]string{
+		"preserved-label":                      "preserved-value",
+		"modified-label":                       "modified-value-2",
+		clusterv1.ClusterNameLabel:             testCluster.Name,
+		clusterv1.MachineControlPlaneLabel:     "",
+		clusterv1.MachineControlPlaneNameLabel: kcp.Name,
+	}
+	kcp.Spec.MachineTemplate.ObjectMeta.Annotations = map[string]string{
+		"preserved-annotation": "preserved-value",  // Keep the annotation and value as is
+		"modified-annotation":  "modified-value-2", // Modify the value of the annotation
+		// Drop "dropped-annotation"
+	}
+	kcp.Spec.MachineTemplate.NodeDrainTimeout = duration10s
+	kcp.Spec.MachineTemplate.NodeDeletionTimeout = duration10s
+	kcp.Spec.MachineTemplate.NodeVolumeDetachTimeout = duration10s
+
+	// Use the updated KCP.
+	controlPlane.KCP = kcp
+	g.Expect(reconciler.syncMachines(ctx, controlPlane)).To(Succeed())
+
+	// Verify in-place mutable fields are updated on the Machine.
+	updatedInplaceMutatingMachine = inPlaceMutatingMachine.DeepCopy()
+	g.Eventually(func(g Gomega) {
+		g.Expect(env.GetAPIReader().Get(ctx, client.ObjectKeyFromObject(updatedInplaceMutatingMachine), updatedInplaceMutatingMachine))
```

Comment:
Do we need the eventually here and below? Seems everything should work without it

Reply:
You are right, since the test uses APIReader we dont need to use an Eventually block. 👍🏼